gufe/components/explicitmoleculecomponent.py

```diff
@@ -14,7 +14,8 @@
 def _ensure_ofe_name(mol: RDKitMol, name: str) -> str:
     """
     Determine the correct name from the rdkit.Chem.Mol and the user-provided
-    name; ensure that is set in the rdkit representation.
+    name; ensure that is set in the rdkit representation. We also perform some sanitation of the name
```

Comment:
My understanding is that this is going to change gufe keys for on read. Is this an issue for anyone? My initial take is "no because you only generate the key on read so it's stable at that point", but probably @dotsdl needs to weigh in on if there's anything alchemiscale wise we need to care about.

Reply:
This won't impact reading data out of an existing `alchemiscale` instance, but it would impact re-use of those components in an `AlchemicalNetwork` that is submitted to a `Scope` with an existing `AlchemicalNetwork` with the same components.

This is generally something that is hard to avoid with new releases, however, due sometimes to stack changes upstream, so I wouldn't consider it a blocker here.